universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -237,28 +229,7 @@ private Result withMessage(Result result, String message) {
 
   private QuestionForm createQuestionForm(Request request, QuestionType questionType)
       throws InvalidQuestionTypeException {
-    switch (questionType) {
-      case ADDRESS:
-        return formFactory.form(AddressQuestionForm.class).bindFromRequest(request).get();
-      case CHECKBOX:
-        return formFactory.form(CheckboxQuestionForm.class).bindFromRequest(request).get();
-      case DROPDOWN:
-        return formFactory.form(DropdownQuestionForm.class).bindFromRequest(request).get();
-      case FILEUPLOAD:
-        return formFactory.form(FileUploadQuestionForm.class).bindFromRequest(request).get();
-      case NAME:
-        return formFactory.form(NameQuestionForm.class).bindFromRequest(request).get();
-      case NUMBER:
-        return formFactory.form(NumberQuestionForm.class).bindFromRequest(request).get();
-      case RADIO_BUTTON:
-        return formFactory.form(RadioButtonQuestionForm.class).bindFromRequest(request).get();
-      case REPEATER:
-        return formFactory.form(RepeaterQuestionForm.class).bindFromRequest(request).get();
-      case TEXT:
-        return formFactory.form(TextQuestionForm.class).bindFromRequest(request).get();
-      default:
-        throw new InvalidQuestionTypeException(questionType.toString());
-    }
+    return QuestionFormBuilder.createFromRequest(request, formFactory, questionType);
```

Comment:
suggestion: Similar to my other comment below. Should we get rid of this `createQuestionForm` method since it's just a one-liner now?

Reply:
Good suggestion! Done.